components/donationFrequency.js

```diff
@@ -1,24 +1,43 @@
 import React from 'react'
 import { Box } from '@chakra-ui/react'
-import { fundraisingEventString } from '../lib/constants'
+import { fundraisingEventTicket, fundraisingEventTicketPatron } from '../lib/constants'
 
-const DonationFrequency = ({ updateFrequency, frequency }) => (
+const DonationFrequency = ({ updateFrequency, frequency, frequenciesAvailable }) => (
   <div className='flex flex-row tf-lato mt1 mb4' role='radiogroup' aria-labelledby='donate-freq'>
     <span id='donate-freq' className='sr-only--text'>Donation Frequency</span>
-    <label className={`tc mt1 pv2 ph1 bn ba b--black ttu pointer flex-auto ${frequency === 'once' ? 'bg-tf-yellow tf-dark-gray' : 'tf-dark-gray bg-white pointer'}`}>
-      <input className='sr-only--input' type='radio' name='frequency' value='once' onInput={updateFrequency} />
-      Give Once
-    </label>
-    <Box width='1px' bg='black' height='auto' marginTop='1' />
-    <label className={`tc mt1 pv2 ph1 bn ba b--black ttu pointer flex-auto ${frequency === 'monthly' ? 'bg-tf-yellow tf-dark-gray' : 'tf-dark-gray bg-white pointer'}`}>
-      <input className='sr-only--input' type='radio' name='frequency' value='monthly' onInput={updateFrequency} />
-      Monthly
-    </label>
-    <Box width='1px' bg='black' height='auto' marginTop='1' />
-    <label className={`tc mt1 pv2 ph1 bn ba b--black ttu pointer flex-auto ${frequency === fundraisingEventString ? 'bg-tf-yellow tf-dark-gray' : 'tf-dark-gray bg-white pointer'}`}>
-      <input className='sr-only--input' type='radio' name='frequency' value={fundraisingEventString} onInput={updateFrequency} />
-      Fall 2022 Event
-    </label>
+    {frequenciesAvailable.includes('once') && (
```

Comment:
Pretty wacky way to construct this component. It shouldn't be necessary to have a custom JSX block for all 4 options. Pain to maintain, pain to read, pain to extend.